### PR TITLE
Higher Learning Rate: 3e-4 for broader loss landscape exploration

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The baseline uses `--lr 2e-4` with Lion optimizer and cosine annealing. This LR was tuned earlier in the research programme when the model had fewer features. Since then, we've added TE coord frame (+6 channels), wake deficit (+2 channels), DCT frequency loss, and multiple augmentations — the loss landscape has changed.

**A higher peak LR (3e-4, 1.5x current)** gives the Lion optimizer larger initial steps, enabling broader exploration of the loss landscape before cosine annealing converges to the final solution. With sign-based updates (Lion), higher LR means each step covers more distance in weight space.

**Why now:** The Lookahead experiment (#2241) showed that CONSTRAINING exploration hurts (all metrics worse). The opposite direction — giving the optimizer MORE freedom early — may help. Higher LR + cosine annealing = more exploration early, same precision late.

**Confidence:** Medium. LR is the most impactful single hyperparameter. 1.5x is a moderate increase that should be safe with cosine annealing (the LR quickly decays). The risk: if 2e-4 was already at the stability boundary, 3e-4 could cause early-training instability.

## Instructions

**No code changes needed.** Change `--lr 2e-4` to `--lr 3e-4`.

### Run 2 seeds

```bash
# Seed 42
cd cfd_tandemfoil && python train.py \
  --agent askeladd --wandb_name "askeladd/lr-3e4-s42" \
  --wandb_group "round19/higher-lr" \
  --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 3e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature

# Seed 73 — identical but --seed 73 --wandb_name "askeladd/lr-3e4-s73"
```

Only change: `--lr 3e-4` (was `2e-4`)

### Report results

Table: p_in, p_oodc, p_tan, p_re for both seeds, 2-seed avg, baseline comparison, W&B run IDs.

Note early training behavior — if the loss spikes or oscillates heavily in the first 10 epochs, 3e-4 may be too high.

## Baseline

| Metric | Baseline | Target to beat |
|--------|----------|----------------|
| p_in   | **11.979** | < 11.98 |
| p_oodc | **7.643**  | < 7.65  |
| p_tan  | **28.341** | < 28.34 |
| p_re   | **6.300**  | < 6.30  |

W&B: `hgml7i2r` (s42), `qic03vrg` (s73)

**Reproduce baseline:**
```bash
cd cfd_tandemfoil && python train.py --agent askeladd --wandb_name "askeladd/baseline" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT askeladd:

## Results

### Higher Learning Rate (3e-4) — 2-seed experiment

| Metric | Baseline | Seed 42 | Seed 73 | 2-seed avg | Δ |
|--------|----------|---------|---------|------------|---|
| **p_in** | **11.979** | 12.690 | 13.324 | **13.007** | **+8.6% ❌** |
| **p_oodc** | **7.643** | 7.935 | 8.177 | **8.056** | **+5.4% ❌** |
| **p_tan** | **28.341** | 29.457 | 28.725 | **29.091** | **+2.6% ❌** |
| **p_re** | **6.300** | 6.528 | 6.339 | **6.433** | **+2.1% ❌** |

**Surface MAE (in-dist):**
| Component | Seed 42 | Seed 73 |
|-----------|---------|---------|
| Ux | 2.788 | 2.987 |
| Uy | 0.832 | 0.898 |
| p  | 12.690 | 13.324 |

**Volume MAE (in-dist):**
| Component | Seed 42 | Seed 73 |
|-----------|---------|---------|
| Ux | 0.527 | 0.570 |
| Uy | 0.199 | 0.205 |
| p  | 13.346 | 14.412 |

### Commands used
```bash
# Seed 42
cd cfd_tandemfoil && CUDA_VISIBLE_DEVICES=0 python train.py \
  --agent askeladd --wandb_name "askeladd/lr-3e4-s42" \
  --wandb_group "round19/higher-lr" --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 3e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature

# Seed 73 — identical but --seed 73 --wandb_name "askeladd/lr-3e4-s73"
```

- **Epochs completed:** 148/500 (both seeds, hit 30-min timeout)
- **W&B run IDs:** `m9t144lx` (seed 42), `bu5ottox` (seed 73)
- **W&B group:** `round19/higher-lr`

### What happened

**The higher LR (3e-4) hurt all metrics uniformly.** Every split got worse — p_in degraded the most at +8.6%, followed by p_oodc (+5.4%), p_tan (+2.6%), and p_re (+2.1%). Both seeds show the same trend, confirming this is not noise.

The hypothesis that broader early exploration would help did not pan out. Instead, the 1.5x LR increase appears to have pushed the optimizer past the sweet spot for this loss landscape. With Lion's sign-based updates, the effective step size increase from 2e-4 → 3e-4 is significant — each update overshoots the fine-grained local optima that the baseline converges to.

The in-distribution splits (p_in, p_oodc) suffered more than the transfer/OOD splits (p_tan, p_re), suggesting the higher LR particularly hurts the model's ability to fit the core training distribution precisely.

**Conclusion:** 2e-4 is well-calibrated for this model + Lion + cosine annealing setup. The current LR is not limiting performance.

### Suggested follow-ups

- **Lower LR (1.5e-4 or 1e-4):** If 3e-4 overshoots, a slightly lower LR might help — though the Lookahead result (constraining exploration hurts) suggests 2e-4 may already be near-optimal.
- **Warmup schedule:** Instead of changing the peak LR, try a warmup phase (e.g. linear warmup for 5-10 epochs) before the cosine annealing kicks in. This gives controlled early exploration without the instability risk.
- **LR per parameter group:** Different LRs for the backbone vs surface refinement head — the refinement head may benefit from a different LR than the main encoder.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2247 because ADVISOR: Closing — all 4 metrics regressed significantly (p_in +8.6%, p_oodc +5.4%, p_tan +2.6%, p_re +2.1%). 3e-4 is clearly past the stability boundary for Lion + this model. LR 2e-4 is well-calibrated. Thank you for the thorough analysis — the confirmation that constraining exploration hurts (Lookahead) while MORE exploration also hurts (higher LR) tells us 2e-4 is at or near the optimal LR for this setup. Adding to dead ends list..
